### DISCUSSION TITLE
Dynamic color : Is this feature possible?

### DISCUSSION THREAD

**Discussion by ralppppy** [tailwindlabs/tailwindcss#3065] [Help] [answered]
Is it possible to have this kind of functionality? Where you will just say "bg-color-#fff" or any hex value, The hex value is really important for dynamic colors. Where a user can customize their colors. For example, change the background of a card.  It's just a thing that I want to add to my project. Currently, I need to manually add colors to the config files. This thing is easily achievable in normal CSS. But I use tailwind because I don't want to leave my HTML.

**Comment by jonnitto** [comment]
Why not use inline styles or javascript (if you need hover states or responsive behavior)? I mean, you could even parse the `bg-color-#fff` and apply the styles on it. 

But: If you want to make it the tailwind style, you could use [Alpine.js](https://github.com/alpinejs/alpine) for that.

**Comment by simonswiss** [comment]
Doing so would break one of the fundamental advantages of Tailwind CSS: **constraints**.

By allowing a user to pass any arbitrary hex color for example, you opt out of the "restricted amount of choices" that makes for consistency in your project.

I don't think this will ever be a thing in Tailwind CSS - but if you're looking to use this workflow, this is basically how https://acss.io/ works.

**Reply by simonswiss** [reply]
Plot twist: I was wrong 🤣

Arbitrary colors like `bg-color-[#0099aa]` are now supported with the new `JIT` engine:

https://tailwindcss.com/docs/just-in-time-mode#arbitrary-value-support

Never say never!

**Comment by cmendes0101** [comment]
I understand this goes away from constraints but I have a need for this. A user is able to select from a color palette as hex to highlight content and this is served from our API which is also used by our mobile apps. 

It would be nice to have the option to place a setting that would allow this to work since we are not able to remove the color palette usage to force into a pre-assigned color list.

**Reply by skttl** [reply]
Supported in 2.1 when using JIT mode https://tailwindcss.com/docs/just-in-time-mode#arbitrary-value-support

**Comment by simonswiss** [comment, marked as answer]
Despite what I said a few comments higher, this is now [a feature of the `JIT` mode](https://tailwindcss.com/docs/just-in-time-mode#arbitrary-value-support) which was recently released! 🎉

So, you can now use `bg-[#0099aa]` to generate a teal-ish background color 👍

**Reply by Mahesh1356** [reply]
Is there any options to send dynamic hex value

**Reply by bdbch** [reply]
@Mahesh1356 not as far as I know. The JIT engine needs to read your files and what kind of values it needs to transform. It's not working with dynamic values. (for example something like `bg-[${themeColor}]`).

What you can do is safelist some classes (which is also not 100% dynamic):
https://tailwindcss.com/docs/optimizing-for-production#safelisting-specific-classes

**Reply by JiahuiChen99** [reply]
> Is there any options to send dynamic hex value

> @Mahesh1356 not as far as I know. The JIT engine needs to read your files and what kind of values it needs to transform. It's not working with dynamic values. (for example something like `bg-[${themeColor}]`).
> 
> What you can do is safelist some classes (which is also not 100% dynamic): https://tailwindcss.com/docs/optimizing-for-production#safelisting-specific-classes

@Mahesh1356 I tried to load dynamic values and it didn't work... I had to write the entire word **bg-[#hex_color]**

Take a look in the discussion thread I opened a few days ago: https://github.com/tailwindlabs/tailwindcss/discussions/7092
